src/__tests__/bim.js

```diff
@@ -0,0 +1,245 @@
+            }
+        },
+    });
+
+    Store.getVocabsItems().forEach((vocab) => {
+        expect(vocab.perkataan).toExist();
+    });
+});
+
+// Test Category in BIM Tab existence in Group Tab
+test("Test 3: Category in BIM Tab should exist according to Category in Group Tab", () => {
+    // Put Category column into array
+    const cat = Store.getGroupCategoryItems().map((item) => item.groupCategory);
+    expect.extend({
+        toHaveOneOf(cat, groupCat) {
+            const pass = cat.includes(groupCat);
+
+            if (pass) {
+                return {
+                    message: () => `All categories matches correctly.`,
+                    pass: true,
+                };
+            } else {
+                return {
+                    message: () =>
+                        `${groupCat} mismatched. Please check the spelling or add the correct category inside the Category column (BIM sheet).`,
+                    pass: false,
+                };
+            }
+        },
+    });
+
+    Store.getVocabItem().forEach((vocab) => {
+        expect(cat).toHaveOneOf(...vocab.groupCategory);
+    });
+});
+
+// Test for Youtube video links
+test("Test 4: Youtube video link is a URL", () => {
+    const isValidHttpUrl = (string) => {
+        let url;
+
+        try {
+            url = new URL(string);
+        } catch (_) {
+            return false;
+        }
+
+        return url.protocol === "http:" || url.protocol === "https:";
+    };
+
+    expect.extend({
+        isValid(uri) {
+            const pass = isValidHttpUrl(uri);
+
+            if (pass) {
+                return {
+                    message: `All links are valid.`,
+                    pass: true,
+                };
+            } else {
+                return {
+                    message: `${uri} is invalid. Please re-copy the valid link and replace.`,
+                    pass: false,
+                };
+            }
+        },
+    });
+
+    Store.getVocabsItems().forEach((url) => {
+        expect(url.video).isValid();
+    });
+});
+
+// Test for SignOfTheDay date format
+test("Test 5:SignOfTheDay date should be written according to standard format", () => {
+    const sotdDate = Store.getVocabsItems()
+        .filter((und) => und.sotd !== undefined)
+        .map((date) => date.sotd.toString());
+    const regEx = /^\d{4}-\d{2}-\d{2}$/;
+
+    sotdDate.forEach((date) =>
+        expect(date).toEqual(expect.stringMatching(regEx))
+    );
+
+    expect.extend({
+        toBeDistinct(sotd) {
+            const pass =
+                Array.isArray(sotd) && new Set(sotd).size === sotd.length;
+            if (pass) {
+                return {
+                    message: () => `SOTD array is unique. No repeated dates.`,
+                    pass: true,
+                };
+            } else {
+                return {
+                    message: () =>
+                        `SOTD array is not unique. There are repeated dates.`,
```

Comment:
You could mention the number of duplicates. Or better yet, print the duplicates in this.